### PR TITLE
[EA-81] Replace SessionFactory with DbSessionFactory

### PR DESCRIPTION
**Pull request by k-joseph**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/emrapi/concept/HibernateEmrConceptDAO.java [hunk 1]
```diff
@@ -53,121 +50,123 @@
  *
  */
 public class HibernateEmrConceptDAO implements EmrConceptDAO {
-
```
Comment (k-joseph) [inline api/src/main/java/org/openmrs/module/emrapi/concept/HibernateEmrConceptDAO.java]: Looks like the final contributor to this file did not format or used a different code for matter than our official openmrs formatter

File: api/src/main/java/org/openmrs/module/emrapi/db/HibernateSingleClassDAO.java [hunk 2]
```diff
@@ -1,70 +1,70 @@
 package org.openmrs.module.emrapi.db;
 
-import org.hibernate.SessionFactory;
-import org.springframework.transaction.annotation.Transactional;
-
 import java.util.List;
 
-public abstract class HibernateSingleClassDAO<T> implements SingleClassDAO<T> {
```
Comment (k-joseph) [inline api/src/main/java/org/openmrs/module/emrapi/db/HibernateSingleClassDAO.java]: Looks like the final contributor to this file did not format or used a different code for matter than our official openmrs formatter

File: omod/src/main/resources/config.xml [hunk 3]
```diff
@@ -12,7 +12,7 @@
         ${project.parent.description}
     </description>
     <updateURL>https://modules.openmrs.org/modules/download/${project.parent.artifactId}/update.rdf</updateURL>
-    <require_version>1.9.8</require_version>
+    <require_version>1.11.3, 1.10.2 - 1.10.*, 1.9.9 - 1.9.*, 1.9.8</require_version>
```
Comment (rkorytkowski) [inline omod/src/main/resources/config.xml]: 1.9.8 should not be here

File: api/src/main/java/org/openmrs/module/emrapi/patient/HibernateEmrPatientDAO.java [hunk 4]
```diff
@@ -75,11 +76,11 @@ public void setEmrApiProperties(EmrApiProperties emrApiProperties) {
 	private Criteria buildCriteria(String query, Criteria criteria) {
 		if (query.matches(".*\\d.*")) {
 			// has at least one digit, so treat as an identifier
-			return new PatientSearchCriteria(sessionFactory, criteria).prepareCriteria(null, query,
+			return new PatientSearchCriteria((SessionFactory) sessionFactory, criteria).prepareCriteria(null, query,
 			    emrApiProperties.getIdentifierTypesToSearch(), true, true, true);
 		} else {
 			// no digits, so treat as a name
-			return new PatientSearchCriteria(sessionFactory, criteria).prepareCriteria(query, null,
+			return new PatientSearchCriteria((SessionFactory) sessionFactory, criteria).prepareCriteria(query, null,
```
Comment (rkorytkowski) [inline api/src/main/java/org/openmrs/module/emrapi/patient/HibernateEmrPatientDAO.java]: You must not cast. Call sessionFactory.getHibernateSessionFactory().

File: api/src/main/java/org/openmrs/module/emrapi/patient/HibernateEmrPatientDAO.java [hunk 5]
```diff
@@ -75,11 +76,11 @@ public void setEmrApiProperties(EmrApiProperties emrApiProperties) {
 	private Criteria buildCriteria(String query, Criteria criteria) {
 		if (query.matches(".*\\d.*")) {
 			// has at least one digit, so treat as an identifier
-			return new PatientSearchCriteria(sessionFactory, criteria).prepareCriteria(null, query,
+			return new PatientSearchCriteria((SessionFactory) sessionFactory, criteria).prepareCriteria(null, query,
```
Comment (rkorytkowski) [inline api/src/main/java/org/openmrs/module/emrapi/patient/HibernateEmrPatientDAO.java]: You must not cast. Call sessionFactory.getHibernateSessionFactory().

**Comment by dkayiwa** [comment]
Closing because it was already done at https://github.com/openmrs/openmrs-module-emrapi/commit/9e30adafd7a25e06312a1385b4f96c70cb1391c3

**Pull request closed without merging**